proposals/p5606.md

```diff
@@ -0,0 +1,160 @@
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Require documentation updates as part of the proposal](#require-documentation-updates-as-part-of-the-proposal)
+    -   [One TODO per document](#one-todo-per-document)
+    -   [Link to the PR instead of the proposal document](#link-to-the-pr-instead-of-the-proposal-document)
+
+<!-- tocstop -->
+
+## Abstract
+
+Require language design proposals to either update the design documents to
+reflect the proposed changes, or add "TODO" comments to mark where those changes
+will be needed, with links back to the proposal. This is intended to ensure that
+the design documentation accurately informs readers about the current language
+design, without excessively burdening the proposal process.
+
+## Problem
+
+The current evolution process allows us to adopt language design changes while
+deferring the corresponding changes in `/docs/design`. We have no process for
+ensuring that those follow-up changes actually happen, and in practice some
+adopted proposals have gone for well over a year without corresponding design
+document changes.
+
+This problem is not limited to proposals for new features, but also applies to
+proposals to change existing, documented features. As a result, the design
+documents are not merely incomplete, but in some cases actually misleading,
+which has led to miscommunication and wasted effort within the Carbon team.
+
+## Proposal
+
+This proposal modifies our evolution process to require proposals to either
+implement the corresponding changes to the design documents, or mark the places
+that will need to be changed with "TODO" comments that point back to the
+proposal. The presence of those comments should help ensure that readers are not
+misled by the design documentation, and the links to the proposals should help
+ensure readers can discover the actual design for a given feature with
+reasonable effort.
+
+The proposal PR also adds "TODO" comments to `/docs/design`, both as an
+illustration of what this policy asks for, and because the policy needs to be
+applied retroactively in order to actually solve the problem. However, these
+changes are necessarily best-effort, because it wasn't feasible for me to fully
+evaluate every proposal against the current state of the docs. In particular, I
+didn't look at proposals numbered below 2000, and I assumed that proposals fully
+updated `/docs/design` if they touched it at all. In addition, I did not add
+"TODO" comments for the terminology changes in [p2964](/proposals/p2964.md),
+because they would be pervasive, and probably add little value for the reader.
+Instead, I filed issue
+[#5599](https://github.com/carbon-language/carbon-lang/issues/5599) to track the
+task of making those changes.
```

Comment:
Should this also be an option we leave open for proposal authors, particularly for widespread but systematic changes that might otherwise require adding TODOs to a lot of different files?

Reply:
Good point, done.